docker/daemon.go

```diff
@@ -101,11 +102,16 @@ func mainDaemon() {
 	// load the daemon in the background so we can immediately start
 	// the http api so that connections don't fail while the daemon
 	// is booting
+	var wg sync.WaitGroup
+	wg.Add(1)
 	go func() {
+		defer wg.Done()
```

Comment:
As discussed on IRC, can't use timeout since there's really no way to know how long the daemon will take to load since this depends on number of images/containers.